### PR TITLE
(iOS) Add support for device name on iOS

### PR DESCRIPTION
**Pull request by BigBalli**
### Motivation and Context

Progresses #130 

### Checklist

- [ ] I've run the tests to see all new and existing tests pass
- [ ] I added automated test coverage as appropriate for this change
- [ ] Commit is prefixed with `(platform)` if this change only applies to one platform (e.g. `(android)`)
- [ ] If this Pull Request resolves an issue, I linked to the issue in the text above (and used the correct [keyword to close issues using keywords](https://help.github.com/articles/closing-issues-using-keywords/))
- [ ] I've updated the documentation if necessary

### CODE REVIEW

**Comment by erisu** [comment]
Can you fill out the PR description, following the template?

And if it is to close a ticket also link the ticket. For example: `Closes #<ticket number>`

Also when I glanced the code, why are you stripping out the non-alphanumeric?

What happens if the device name is in another language such as Japanese, Chinese, etc?

**Comment by BigBalli** [comment]
Sorry, first time contributing to a Cordova repo.

- I only tested the changes locally, didn't run specific automated tests (not sure what testing environment/workflow you usually use.)
- prefixed the commit with platform.
- not related to an existing issue
- didn't update the documentation

Please let me know how to proceed,
G

File: src/ios/CDVDevice.m [hunk 1]
```diff
@@ -86,10 +86,14 @@ - (void)getDeviceInfo:(CDVInvokedUrlCommand*)command
 - (NSDictionary*)deviceProperties
 {
     UIDevice* device = [UIDevice currentDevice];
+	//strip non-alphanumeric from device name
+    NSCharacterSet *charactersToRemove = [[NSCharacterSet alphanumericCharacterSet] invertedSet];
+    NSString *deviceName = [[[device name] componentsSeparatedByCharactersInSet:charactersToRemove] componentsJoinedByString:@""];
```
Comment (breautek) [inline src/ios/CDVDevice.m:91]: Apple's documentation for [UIDevice->name](https://developer.apple.com/documentation/uikit/uidevice/1620015-name?language=objc) states

> The value of this property is an arbitrary alphanumeric string that is associated with the device as an identifier. For example, you can find the name of an iOS device in the General > About settings.

So is scanning for non-alphanumeric characters really necessarily? Apple specifically states it's an alphanumeric string.

Comment (BigBalli) [reply src/ios/CDVDevice.m:91]: True, Apple's documentation states "alphanumeric string" but from experience it is not so.
You can go change your device's name to anything you want (ie no forbidden characters).

Comment (breautek) [reply src/ios/CDVDevice.m:91]: I see, thanks for the clarification

**Comment by jcesarmobile** [comment]
Going to close since starting on iOS 16 this is considered a sensitive piece of information and doesn’t return the value anymore unless you request a special entitlement to Apple, so the addition has limited usefulness for most users

https://developer.apple.com/documentation/uikit/uidevice/1620015-name

**Pull request closed without merging**